### ISSUE TITLE
Compilation ouput has undefined properties (__generator function param _b.trys is undefined)

### ISSUE THREAD

**Issue by orkanone** [microsoft/TypeScript#40897]
**TypeScript Version:**  4.1.0-dev.20201001

**Search Terms:** 'push' of undefined

**Code:**
Checkout https://github.com/orkanone/typescript-compilation-error

```
tsc
node dist/index.js
```

**Expected behavior:**
Should log ` { name: 'A', vendor: 'shop1', category: 'one', bagel: true }`

**Actual behavior:**
```
(node:16280) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'push' of undefined
    at .\typescript-generation-error\dist\index.js:110:45
    at step (.\typescript-generation-error\dist\index.js:43:23)
    at Object.next (.\typescript-generation-error\dist\index.js:24:53)
    at step (.\typescript-generation-error\dist\index.js:28:139)
    at Object.next (.\typescript-generation-error\dist\index.js:24:53)
    at .\typescript-generation-error\dist\index.js:18:71
    at new Promise (<anonymous>)
    at __awaiter (.\typescript-generation-error\dist\index.js:14:12)
    at showProduct (.\typescript-generation-error\dist\index.js:92:12)
    at Object.<anonymous> (.\typescript-generation-error\dist\index.js:175:1)
(node:16280) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:16280) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by RyanCavanaugh** [comment]
@rbuckton I think we have a similar bug on this already, but can't confirm it's an exact duplicate

**Comment by incomingstick** [comment]
We were having this issue too. `TypeError: Cannot read property 'push' of undefined` inside of a compiled main.js file at a `_b.trys.push()` call where `_b.trys` is undefined.

I took a look at your example project and noticed your comments at the bottom of the index.ts file:
```
// The code works when:
// no async await - request returns an object, not a promise (l.40, l.54)
// for..in loop instead of for..of (53)
// products are not filtered (l.51)
// no optional chaining in if condition (l.47)
```

I tested these various things in my code, and removing a set of `optional chaining in [an] if condition` resolved this issue for me.
